### PR TITLE
Add support for using oauth

### PR DESCRIPTION
**Pull request by farcaller**
Follows up on the oauth support in https://github.com/tailscale/tailscale/pull/7393.

### CODE REVIEW

**Review by shayne — APPROVED** [review]

**Comment by shayne** [comment]
cc/ @DentonGentry who reviewed tailscale/tailscale#7393

File: action.yml [hunk 1]
```diff
@@ -6,7 +6,13 @@ inputs:
     required: true
   api-key:
     description: "Tailscale API key"
-    required: true
+    required: false
+  oauth-id:
```
Comment (DentonGentry) [inline action.yml:10]: Could you make this `oauth-client-id` to match https://github.com/tailscale/github-action/blob/7a0b30ed3517c2244d1330e39467b95f067a33bd/action.yml#L14 ?

I understand this means the YAML takes a `oauth-client-id` argument while directly invoking `gitops-pusher` will take a `--oauth-id=...` argument, but I think commonality between the different GitHub Actions YAML arguments is more important than matching the `gitops-pusher` CLI flag.

Comment (shayne) [inline action.yml:10]:
```suggestion
  oauth-client-id:
```

File: action.yml [hunk 2]
```diff
@@ -19,6 +25,8 @@ runs:
   image: "Dockerfile"
   env:
     TS_API_KEY: "${{ inputs.api-key }}"
+    TS_OAUTH_ID: "${{ inputs.oauth-id }}"
```
Comment (shayne) [inline action.yml:28]:
```suggestion
    TS_OAUTH_ID: "${{ inputs.oauth-client-id }}"
```

**Comment by DentonGentry** [comment]
We'll merge this in a bit, I want to queue up a PR to add something similar to https://github.com/tailscale/github-action/blob/7a0b30ed3517c2244d1330e39467b95f067a33bd/action.yml#L52-L57 (but not expecting a Tags argument to be present).

File: action.yml [hunk 3]
```diff
@@ -6,7 +6,13 @@ inputs:
     required: true
   api-key:
     description: "Tailscale API key"
-    required: true
+    required: false
+  oauth-client-id:
+    description: "tailscale OAuth ID"
+    required: false
+  oauth-secret:
+    description: "tailscale OAuth Secret"
```
Comment (pmocek) [inline action.yml:14]: nit: capitalize proper noun
```suggestion
    description: "Tailscale OAuth ID"
    required: false
  oauth-secret:
    description: "Tailscale OAuth Secret"
```

**Comment by tobiasehlert** [comment]
Is this maybe something that @shayne or @DentonGentry might want to merge? :)
Would be great to have this implemented soon!

**Comment by willnorris** [comment]
I've rebased this onto main to pick up some other changes, as well as updated it to use a new tag in the tailscale/tailscale repo that contains a small fix to how gitops-pusher was checking credentials.  I'll squash these and push them to your branch momentarily and then merge.

**Review by willnorris — APPROVED** [review]

**Pull request merged**